pkg/controller/petset/pet.go

```diff
@@ -165,20 +165,15 @@ type apiServerPetClient struct {
 }
 
 // Get gets the pet in the pcb from the apiserver.
-func (p *apiServerPetClient) Get(pet *pcb) (*pcb, bool, error) {
-	found := true
+func (p *apiServerPetClient) Get(pet *pcb) (*pcb, error) {
```

Comment:
It set `found` to `false` only if client receive an "Not Found" error, while keep found to true if client receive other type errors(such as 403).

In other words, I think this function's 2nd returned value should always be **false** when err is not nil, which means we didn't find what we want as Get request failed.

BTW, I don't think it's a good way to reset error to nil by hand here